### ISSUE TITLE
[web component] Menu

### ISSUE THREAD

**Issue by alizedebray** [swisspost/design-system#5628]
### 📝 Description

Used to present a list of actions, commands, or navigation links.
https://www.figma.com/design/JIT5AdGYqv6bDRpfBPV8XR/Foundations---Components-Next-Level?node-id=13997-36408&m=dev

### ♿ Accessibility

- Links, buttons and other focussable elements should be keyboard accessible by pressing up and down arrows when the menu is open.
- Make sure the menu trigger has a valid accessible name

### 💻 Proposed implementation

```html
<!-- Trigger -->
<post-menu-trigger for="my-id">
  My content
</post-menu-trigger>

<!-- Dropdown -->
<post-menu id="my-id">
  <p slot="header">
    My menu header
  </p>
  <post-menu-item>
    <a href="#">My Link</a>
  </post-menu-item>
  <post-menu-item>
    <button type="button">My Button</button>
  </post-menu-item>
  <p slot="footer">
    My menu footer
  </p>
</post-menu>
```

### 🔗 References

- Menu button accessibility documentation: https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/

### 📃 Tasks

- [x] #5753
- [ ] Use tokens for the styles
- [ ] Create storybook documentation
- [ ] Add functional tests
- [ ] Add integration tests for React
- [ ] Add integration tests for Angular
- [ ] Add visual regression tests
- [ ] Design review

**Comment by alizedebray** [comment]
A styled menu was needed in order to develop the #5195. 
As a result, the `post-menu` component already includes style but they are not yet tokenized. Because of this, the component is still documented as a raw component.

This ticket aims to finalize the styles using design tokens and make the component fully available for all users.
